pytorch_lightning/trainer/trainer.py

```diff
@@ -429,7 +428,20 @@ def _setup_on_init(
         else:
             self.num_sanity_val_steps = num_sanity_val_steps
 
-    def _setup_fit(self, model, train_dataloader=None, val_dataloaders=None, datamodule=None) -> None:
+        self.num_sanity_val_batches = []
```

Comment:
Can't this be set within a connector ?

Reply:
It could but our intention was to be more careful how we initialize and change the state of trainer from outside. If the initialization gets scattered across connectors it becomes unclear at which stage a certain attribute is fully initialized and ready to be used. 
@justusschock thoughts?

Reply:
Personally, I'd prefer to keep this here for the same reasons @awaelchli mentioned. 
In general I don't like the pattern that we set (previously undefined) attributes to the trainer from other classes. IMO these should almost always be owned by the respective class and only have a property on the trainer forwarding these if necessary